src/validation.cpp

```diff
@@ -2537,6 +2537,9 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
         return false;
     }
 
+    int nStopAtHeight = GetArg("-stopatheight", DEFAULT_STOPATHEIGHT);
+    if (nStopAtHeight && pindexNewTip->nHeight >= nStopAtHeight) StartShutdown();
```

Comment:
Yes, like 2486. I missed that somehow. `pindexNewTip` is not guaranteed null. Sorry about misleading response.